packages/plugin-constraints/sources/Constraints.ts

```diff
@@ -27,6 +27,54 @@ const DEPENDENCY_TYPES = [
   DependencyType.PeerDependencies,
 ];
 
+class Session {
+  private readonly session: pl.type.Session;
+
+  public constructor(project: Project, source: string) {
+    this.session = pl.create();
+    linkProjectToSession(this.session, project);
+
+    this.session.consult(source);
+  }
+
+  private fetchNextAnswer() {
+    return new Promise<pl.Answer>(resolve => {
+      this.session.answer((result: any) => {
+        resolve(result);
+      });
+    });
+  }
+
+  private async accumulateAnswers() {
```

Comment:
Just thinking out loud, maybe it would make sense later to turn this into an async generator instead of a buffer so that we can print results before having found all the answers 🤔

Reply:
That's the road I went down first, but async generators are not supported in Node 8 without a flag so I had to undo that.

Reply:
TypeScript seems to [transpile them](https://www.typescriptlang.org/play/index.html#src=async%20function*%20foo()%20%7B%0D%0A%7D) on the fly (in a very ugly code, granted 😄).

Reply:
I am an idiot, I completely forgot typescript transpiles newer features. In that case I'm definitely using an async generator.